Use XDG_CONFIG_HOME as first XKB search path

whot opened the pull request:
Use $XDG_CONFIG_HOME/xkb as the primary lookup path for XKB rules. Same
motivation as in 3a91788d9254b, however the XDG directories are more standard
and recommended these days than application-specific dotfiles.

The XDG spec says to fall back to $HOME/.config where XDG_CONFIG_HOME is not
set so we implement that behavior as well.

Fixes #112

bluetech commented:
Thanks @whot.

The implementation is a bit confused though, it calls `home = secure_getenv("HOME");` twice and adds `~/.xkb` to the search path twice.

I also think that in terms of reading the code, it will be clearer if the `~/.config/` fallback is in an `else` block of the `XDG_CONFIG_HOME` check. If it works out.

whot commented:
> adds ~/.xkb to the search path twice.

doh, that whole block should've been removed, copy/paste error. Changed and updated as requested with a follow-up patch that checks for the various include paths to be correct/present.

whot commented:
Ok, test errors are fixed, cause ended up that on macos the default xkb path does not exist - but it appears to exist on the others. Something to keep in mind maybe.

bluetech commented:
Thanks!
